app/src/behaviors/behavior_sensor_rotate_common.c

```diff
@@ -5,48 +5,92 @@
 #include <zephyr/kernel.h>
 
 #include <zmk/behavior_queue.h>
+#include <zmk/virtual_key_position.h>
 
 #include "behavior_sensor_rotate_common.h"
 
 LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
 
-int zmk_behavior_sensor_rotate_common_trigger(struct zmk_behavior_binding *binding,
-                                              const struct device *sensor,
-                                              struct zmk_behavior_binding_event event) {
+int zmk_behavior_sensor_rotate_common_data(struct zmk_behavior_binding *binding,
+                                           struct zmk_behavior_binding_event event,
+                                           const struct zmk_sensor_config *sensor_config,
+                                           size_t channel_data_size,
+                                           const struct zmk_sensor_channel_data *channel_data) {
     const struct device *dev = device_get_binding(binding->behavior_dev);
-    const struct behavior_sensor_rotate_config *cfg = dev->config;
+    struct behavior_sensor_rotate_data *data = dev->data;
+
+    const struct sensor_value value = channel_data[0].value;
+    int triggers;
+    int sensor_position = ZMK_SENSOR_POSITION_FROM_VIRTUAL_KEY_POSITION(event.position);
+
+    // Some funky special casing for "old encoder behavior" where ticks where reported in val2 only,
+    // instead of rotational degrees in val1.
+    if (value.val1 == 0) {
```

Comment:
Doesn't the new encoder driver report degrees in val1 and fractional degrees in val2? If so, won't that mean that if I position my knob at exactly 0 degrees and then rotate it by less than one degree, it will generate somewhere up to a million triggers?

As much as I'd like to maintain backwards compatibility, as I'm reading through this, I'm feeling more and more like it would be simpler to just add forced compiler errors if you haven't set `steps` or `triggers_per_rotation`, and have the error messages link to ZMK documentation describing how to set them (or describe inline if we can find a way to make that short enough).

Reply:
I don't think we *yet* have any encoders in use that have that high a resolution, so I think the risk here is low. I do think we need to get folks migrated as quickly as possible, but would rather not hard break right away. I'd imagine:

* Merge this PR, but include config docs and blog post outlining the config changes needed
* Add `#warning` in the code that would trigger when using the old deprecated prop(s)
* Roll this out, announce in Discord, social media, etc.
* Give it some time (month?) then remove the old code/configs.

Thoughts?

Reply:
I think if we just use a `#warning`, almost nobody will ever see that (why would you go looking through the logs on your GH actions for a _successful_ build?). Anyone who's not watching Discord etc. will just see a hard break later anyways.

Reply:
> I think if we just use a `#warning`, almost nobody will ever see that (why would you go looking through the logs on your GH actions for a _successful_ build?). Anyone who's not watching Discord etc. will just see a hard break later anyways.

I think just because we might miss *some* folks with our announcements/warnings, doesn't mean we shouldn't make the effort to reduce it as much as we can. Agreed that the warning might be a waste of time though, we'll get more bang for our buck w/ blog post that's posted to Twitter, Discord, etc. to catch as many as we can that way.

Reply:
I think as long as we clearly document in the code that this needs to be changed, and we plan to force a compiler error when we eventually change it, then that would be fine.

Reply:
Added a blog post to document the changes: https://deploy-preview-1039--zmk.netlify.app/blog/2023/06/10/encoder-refactors

Thoughts?